File: solita-dev-academy-2023-client/src/Models/DetailedStation.ts
import Station from "./Station";
import PopularStation from "./PopularStation";

export default interface DetailedStation extends Station {
    DepartureCount: number;
    ReturnCount: number;
    DepartureDistanceAverage: number;
    ReturnDistanceAverage: number;
    TopDestinationStations: PopularStation[];
    TopOriginStations: PopularStation[];
}

File: webapi/webapi_library/Models/DetailedStation.cs
namespace webapi_library.Models
{
    public class DetailedStation : Station
    {
        public int DepartureCount { get; set; }
        public int ReturnCount { get; set; }
        public double DepartureDistanceAverage { get; set; }
        public double ReturnDistanceAverage { get; set; }
        public IEnumerable<PopularStation> TopDestinationStations { get; set; } = new List<PopularStation>();
        public IEnumerable<PopularStation> TopOriginStations { get; set; } = new List<PopularStation>();
    }
}
